### ISSUE TITLE
Difference Between Exporting via Terminal and UI in Godot

### ISSUE THREAD

**Issue by nongvantinh** [godotengine/godot#99208]
### Tested versions

Godot 4.4.dev4.mono

### System information

Ubuntu 24.04

### Issue description

I'm exporting a project using Docker, which invokes Godot through the terminal with the following command:

```bash
godot.4.4.dev4.mono "path-to-my-project.godot" --export-release "Headless" --headless --quit
```

However, when I run the app, it doesn't behave as expected. I then tried exporting through the Godot UI with the same configuration, and the output binary worked as expected. I noticed that when exporting via the UI, there is an additional `app-name.sh` file. However, the primary difference lies in the binary itself. The binaries generated from the terminal and UI exports have different underlying instructions.

![image](https://github.com/user-attachments/assets/fa80190d-0c1a-45d5-8752-c1ca31892fb0)

### Steps to reproduce

```bash
godot.4.4.dev4.mono "path-to-my-project.godot" --export-release "Headless" --headless --quit
```

### Minimal reproduction project (MRP)

I don't know the exact step that causes this, but my project contains assets and C# scripts. It doesn't have this problem with a fresh project, though.

**Comment by fire** [comment]
Try using the `--headless` option.

**Comment by nongvantinh** [comment]
Yeah, that's what I did, and it's not working. I have a full-time job to keep, and my app still needs refinement before deploying it to the cloud, so I don't have time to look at the code. The point is, the export project should not have this kind of inconsistency. Both the UI and the terminal should invoke the same function.

**Comment by fire** [comment]
I haven't thoroughly investigated this due to a lack of funding, but suspicious optimizations in the dummy rendering server that zeroed the texture in the past have broken things.

**Comment by nongvantinh** [comment]
I confirm that adding the `--headless` option or removing it doesn't make any difference regardless of the platform. Therefore, exporting through the UI has a different workflow, and exporting through the terminal has a different workflow.

The two command below yield the same bug with the binary:

```
 ./Godot_v4.4-dev_mono_winx86_64.exe "/d/GodotProjects/azheir/azheir/project.godot" --export-release Headless --quit
 ./Godot_v4.4-dev_mono_winx86_64.exe "/d/GodotProjects/azheir/azheir/project.godot" --headless --export-release Headless --quit
```

**Comment by fire** [comment]
If you can provide a minimal sample, it would help us debug the most; if not, a non-minimal sample would work, too, but people value their project's privacy.
